### PR TITLE
Generate a `chainList.json` and `chainList.toml` file

### PR DESCRIPTION
**Pull request by geoknee**
Closes https://github.com/ethereum-optimism/client-pod/issues/839

Replaces `chainIds.json` with `chainList.json` and `chainList.toml` which contain richer information.

~Drive-by change: transcribe all code which generates json files and remove all Javascript code from the repo.~ reverted

### CODE REVIEW

File: superchain/configs/chainList.json
Comment (mds1) [inline superchain/configs/chainList.json:9]: Have we considered approximately mirroring the JSON structure returned by https://chainid.network/chains.json or other similar endpoints for https://github.com/ethereum-lists/chains? For example, this would result in the following changes:
- `rpc` and `explorer` become arrays, which is nice so we don't enshrine a single provider
- Use the `parent` key to specify the appropriate L1
- All new new fields are left untouched

```json
  {
    "name": "Base Sepolia",
    "identifier": "sepolia/base",
    "chainId": 84532,
    "rpc": ["https://sepolia.base.org"],
    "explorers": ["https://sepolia-explorer.base.org"],
    "superchain_level": 2,
    "parent": { ... }
  },
```

Comment (geoknee) [reply superchain/configs/chainList.json]: I tried that out here [30701d3](https://github.com/ethereum-optimism/superchain-registry/pull/265/commits/30701d3369cc2decfc839ec7e27cc630d13b388a) LMKWYT. 

My general comments are:
* we don't support multiple RPCs or explorers in the actual underlying registry structures at the moment (if we like this change, probably we should do that)
* we don't scrape any bridge URLs to put in the "parent" field at the moment
* we aren't using the chain agnostic identification scheme https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md as the ethereum-list repo does
* I hardcoded the parent type as "L2" for now.

Comment (mds1) [reply superchain/configs/chainList.json]: Sounds good. Not a blocker for me, just wanted to flag it as an option since it's possible e.g. MetaMask and others already have scripts to parse that JSON structure. 

I do think it's ok to make RPC and explorers arrays in the output though, even though it's not supported as such in the input. Will defer to you to resolve :)

File: superchain/internal/codegen/main.go
Comment (sebastianst) [inline superchain/internal/codegen/main.go:109]: This new address writer sets all non-existent contracts to the zero address. Non-existent contracts seem to have been omitted before, which I think is better?

File: superchain/internal/codegen/main.go
Comment (sebastianst) [inline superchain/internal/codegen/main.go:103]: nit: give it some imperative name, like
```suggestion
func generateAddresses() {
```

File: superchain/configs/chainList.toml
Comment (sebastianst) [inline superchain/configs/chainList.toml:10]: Interesting, didn't know that toml can be written with such indentation. The [official TOML website](https://toml.io/en/v1.0.0#array-of-tables) doesn't indent. But I guess since it's just produced by the toml go package, it's also correct.

**Review by sebastianst — COMMENTED** [review]
I guess even though all js code is removed, we still want to generate the node package (CI job `publish-bot:`)?

File: .github/CODEOWNERS
Comment (geoknee) [reply .github/CODEOWNERS:20]: Correct.

**Comment by geoknee** [comment]
> I guess even though all js code is removed, we still want to generate the node package (CI job `publish-bot:`)?

I suppose so -- I don't see any trigger for that workflow nor published package on NPM yet. Still, I put the manifest back in [bf4ceca](https://github.com/ethereum-optimism/superchain-registry/pull/265/commits/bf4ceca243427407909035575c6cc528b1a046af) and we can double check offline about the future of the npm package.

File: superchain/internal/codegen/main.go
Comment (geoknee) [reply superchain/internal/codegen/main.go]: [9ba3c38](https://github.com/ethereum-optimism/superchain-registry/pull/265/commits/9ba3c38b2652497c3d2c8360d69fc3ef58bd25f7)[9ba3c38](https://github.com/ethereum-optimism/superchain-registry/pull/265/commits/9ba3c38b2652497c3d2c8360d69fc3ef58bd25f7)

File: superchain/internal/codegen/main.go
Comment (geoknee) [reply superchain/internal/codegen/main.go]: I spent some time on this in the end. The zero addresses being written does stand out somewhat. It is actually out of scope for this PR, so I have reinstated the existing behaviour to unblock us.

I think it is worth coming back to, though. Thoughts and solutions I discarded for now:

1. Writing out zeros for "missing" addresses reflects the behaviour of the Go bindings. I wonder if we should use a pointer to an address inside `AddressList`, especially for contracts which we don't expect to always be there (i.e. optional fields). It seems to better reflect the underlying data -- especially if we ever got into a situation where a zero address was actually a meaningful value for us, it is a meaningful value in the EVM. I actually tried this type change out but it quickly snowballs and is a breaking change for downstream packages. 

2. The `omitempty` directive only works for "empty" fields which are being marshalled, and `[20]byte{}` is not considered empty. It is possible to get around this by writing custom marshalling code, but it is pretty nasty and lots of boilerplate.

File: .circleci/config.yml
Comment (geoknee) [reply .circleci/config.yml:108]: I have added instructions to the readme, it should be invoked after running the add-chain tooling and before raising a PR with the new chain data.

Comment (geoknee) [reply .circleci/config.yml:108]: [b05ebc0](https://github.com/ethereum-optimism/superchain-registry/pull/265/commits/b05ebc09709939500849971cf2524cd5c01dc830)[b05ebc0](https://github.com/ethereum-optimism/superchain-registry/pull/265/commits/b05ebc09709939500849971cf2524cd5c01dc830)[b05ebc0](https://github.com/ethereum-optimism/superchain-registry/pull/265/commits/b05ebc09709939500849971cf2524cd5c01dc830)

File: superchain/internal/codegen/main.go
Comment (geoknee) [reply superchain/internal/codegen/main.go]: Good catch, this was an oversight on my part. Fixed it here [9a34302](https://github.com/ethereum-optimism/superchain-registry/pull/265/commits/9a343023a12f02756db60549b99337411d441b93)

File: README.md
Comment (sebastianst) [inline README.md:99]: It's now 4. 7. 6. instead of 4. 5.

**Review by sebastianst — APPROVED** [review]

File: superchain/internal/codegen/main.go
Comment (sebastianst) [reply superchain/internal/codegen/main.go]: ok bummer that it created so much complexity...

File: README.md
Comment (geoknee) [reply README.md]: [0fd0931](https://github.com/ethereum-optimism/superchain-registry/pull/265/commits/0fd093155a954bd274d400ea169071e6acd4ad61)

**Pull request merged**
